### ISSUE TITLE
landsat project values +/- 1 with shp/mask

### ISSUE THREAD

**Issue by mcorbiere** [matthewhanson/gips#16]
first i reprocesss the tile

mcorbier@nile:/titan/projects/cyanomap/data/fieldwork2014/lakes_visited$ landsat process -t 013030 -d 2014-276 -p ref --overwrite GIPS Landsat utility v0.8.0b4 Processing 1 files: ref -> 013030_2014276_LC8_ref.tif: processed in 0:01:08.158458 Completed processing in 0:01:48.709016

next i project the tile (i projected acca after this)

mcorbier@nile:/titan/projects/cyanomap/data/fieldwork2014/lakes_visited$ landsat project -t 013030 -d 2014-276 -p ref --datadir test GIPS Landsat utility v0.8.0b4 Processing 1 files: ref Completed processing in 0:00:00.000143 Creating GIPS project test/TILEID Dates: 1 dates (2014-10-03 - 2014-10-03) Products: ref Creating 013030_2014276_LC8_ref.tif Completed GIPS project in 0:00:35.544166

then i project the same tile with a shapefile that covers a small part of it (i also projected acca after this)

mcorbier@nile:/titan/projects/cyanomap/data/fieldwork2014/lakes_visited$ landsat project -s crystallake.shp -d 2014-276 -p ref --nowarp --datadir test GIPS Landsat utility v0.8.0b4 Processing 1 files: ref Completed processing in 0:00:00.000106 Creating GIPS project test Dates: 1 dates (2014-10-03 - 2014-10-03) Products: ref Completed GIPS project in 0:00:00.650664

then i mask both of these images with their acca image

mcorbier@nile:/titan/projects/cyanomap/data/fieldwork2014/lakes_visited$ gips_mask test --pmask acca GIPS (v0.8.0b4): Mask (v1.0.0) Masking files from 2014-10-03

mcorbier@nile:/titan/projects/cyanomap/data/fieldwork2014/lakes_visited$ gips_mask test/013030 --pmask acca GIPS (v0.8.0b4): Mask (v1.0.0) Masking files from 2014-10-03

Now I have four images; tile ref, tile ref-masked, shape ref, shape ref-masked

Tile ref-masked is 1 value lower for every band than tile ref.

shape ref is 1 value higher than tile ref for bands 2-8 (taking the shifted pixels into account)

shape ref-masked is one value lower than shape ref for every band. (so it seems to be back to the original values of tile ref, besides band 1 which is one value lower)

Summary:

tile ref: 0 tile ref-masked: -1 (from tile ref) 

shape ref: +1 (except band 1) (from tile ref) 

shape ref-masked -1 (from shape ref)

rad-toa, volref, ref-toa have the same issue. dn seems to not have this issue.

**Comment by ircwaves** [comment]
Only because it's old, @mcorbiere, does this still happen?

**Comment by mcorbiere** [comment]
Yes, still an issue. The value of +/- 1 is before the scale is applied (1E-4 in this case).

mcorbier@rio:/atlas/projects/cyanomap/data/fieldwork2014/lakes_visited/kelp$ gips_project landsat -s ../crystallake.shp -d 2014-276 -p ref acca

mcorbier@rio:/atlas/projects/cyanomap/data/fieldwork2014/lakes_visited/kelp$ gips_mask ./ -p ref --pmask acca

![image](https://cloud.githubusercontent.com/assets/10674698/12364848/17f017bc-bb9f-11e5-98eb-be46098e3aba.png)
